Rebased PR: add docker compose for installing a chatmail relay

missytake opened the pull request:
Thanks @Keonik1 for #614 :) I had a lot of fun with `git rebase` to break it up into small, reviewable chunks. It's still a lot, but doable :)

Before we can merge this, we need to merge:

- [ ] https://github.com/chatmail/relay/pull/661
- [ ] https://github.com/chatmail/relay/pull/660
  - [ ] which depends on https://github.com/chatmail/relay/pull/659
- [ ] https://github.com/chatmail/relay/pull/657
- [ ] https://github.com/chatmail/relay/pull/656

Additional related PRs:

- https://github.com/chatmail/relay/pull/663 - that one is maybe necessary, but I don't think so
- https://github.com/chatmail/relay/pull/662

With this PR, checking DNS records should already work; Getting tests to run with a docker instance is still a bit of work (see the https://github.com/chatmail/relay/compare/docker-rebase...docker-tests branch for first steps).

So for now this will be an experimental feature which we can tell people to try out if they ask for it. But in the long term we need a separate CI machine for testing the docker setup, to be sure nothing breaks when we make changes.

Keonik1 commented on docs/DOCKER_INSTALLATION_RU.md:
````diff
@@ -0,0 +1,177 @@
+```shell
+cp ./docker/docker-compose-default.yaml docker-compose.yaml
+```
+
````
same

```suggestion
2. Скопировать `./docker/example.env` и переименовать в `.env`. Здесь хранятся переменные, которые используются в `docker-compose.yaml`.
```shell
cp ./docker/example.env .env
```\
```

Keonik1 commented on docker/files/setup_chatmail_docker.sh:
```diff
@@ -0,0 +1,78 @@
+
+### MAIN
+
+if [ "$DEBUG_COMMANDS_ENABLED" == "true" ]; then
```
```suggestion
if [ "$DEBUG_COMMANDS_ENABLED" == "true" ]; then
```

Keonik1 commented on docker/files/setup_chatmail_docker.sh:
```diff
@@ -0,0 +1,78 @@
+
+# TODO: Move to debug_commands after git clone is moved to dockerfile. 
+git config --global --add safe.directory /opt/chatmail
+if [ "$RECREATE_VENV" == "true" ]; then
```
```suggestion
if [ "$RECREATE_VENV" == "true" ]; then
```

Keonik1 commented on docker/files/setup_chatmail_docker.sh:
```diff
@@ -0,0 +1,78 @@
+./scripts/cmdeploy init --config "${INI_FILE}" $INI_CMD_ARGS $MAIL_DOMAIN
+bash /update_ini.sh
+
+./scripts/cmdeploy run --ssh-host docker
```
```suggestion
./scripts/cmdeploy run --ssh-host docker
```

missytake commented:
> I created an issue with the main problem.
> 
> #694

I added a commit for this :) 

Also, I rebased this on main, so we can try it out again. It's a bit late now for me to re-deploy it, but I'll check tomorrow whether it still works as expected :)

j4n commented:
Closing this as superceded, current efforts in #903 / https://github.com/chatmail/docker
